lib/tasks/sync_users.rake

```diff
@@ -0,0 +1,187 @@
+  # then this is a user account manually fixed by the Library Outreach team.
+  # call like this:  RAILS_ENV=local rake check_user_fixes:check_mismatch['db/migrate/20181128_sierra_mln_user_accounts.csv',2,1,'23333090060508']
+  desc "Check manually made account fixes"
+  task :check_user_fixes, [:file_name, :start, :limit, :barcode]  => :environment do |t, args|
+    # if barcode is set, then only checks that barcode for Sierra-MLN mismatch
+    args.with_defaults(:file_name => nil, :start=> 0, :limit => 0, :barcode => nil)
+
+    csv_start = args.start.to_i
+    csv_limit = args.limit.to_i
+
+    csv_barcode = args.barcode
+
+    # Example:  'db/migrate/20181128_sierra_mln_user_accounts.csv'
+    csv_path = args.file_name
+
+    # \u03EA is a unicode character that will never happen in our Sierra output file.
+    # we need this, because the Sierra output has some junk data that's making the CSV gem trip up.
+    # however, this means that we'll now need to strip the double quotes from around the ingested field values ourselves.
+    CSV.foreach(csv_path, col_sep: "|", quote_char: "\u03EA") do |line|
+      # don't process the header row
+      next if $. == 1
+
+      # if we're only reading a few lines of the csv, then see if we can stop, before reading any further
+      next if (csv_start > 0 && $. < csv_start)
+      break if (csv_limit > 0 && $. >= (csv_start + csv_limit))
+
+      # "P BARCODE"|"EMAIL ADDR"|"PIN"|"EXP DATE"|"PCODE3"|"P TYPE"|"TOT CHKOUT"|"HOME LIBR"|"MBLOCK"|"PCODE4"|"PATRN NAME"|"ADDRESS"|"CREATED(PATRON)"|"UPDATED(PATRON)"
+      # Example:  "23333106701234"|"name@gmail.com"|"Tn/jC7sHXQpfw"|"10-01-2018"|"1"|"153"|"4"|"ea   "|"-"|"895"|"LASTNAME, FIRSTNAME"|"123 MAIN ST, NY 10001"|"05-20-2009 15:23"|"07-02-2018"
+      (barcode, email, pin, expiration, pcode3, ptype, total_checkouts, home_library, manual_block, pcode4, name, address, sierra_created, sierra_updated) = line
+
+      # inore bad data, and move on to the next line
+      next if (barcode.nil?)
+
+      # clean off any leading/trailing spaces and double quotes
+      barcode = barcode.strip.downcase
+      barcode = barcode.chomp('"').reverse.chomp('"').reverse
+      sierra_created = sierra_created.chomp('"').reverse.chomp('"').reverse
+
+      # if a barcode was passed, we're looking for that barcode, and should stop processing once we find it.
+      if (!csv_barcode.nil? && csv_barcode == barcode)
+        csv_start = $.
+        csv_limit = 1
+      end
+
+      # is there a user in the mln database that matches this user from Sierra?
+      mln_user = User.find_by_barcode(barcode)
+      if !mln_user.nil?
+        # Yes, there is.  Now see if the user was created in MLN over the last month, while being old in Sierra.
+
+        # was the user created after a month ago in mln database?
+        if (mln_user.created_at > 1.month.ago)
```

Comment:
Is this rake task going to run automatically at the same time and day of month?

Reply:
No, just once.  And I actually won't be running it on production -- I already ran it on a copy of production locally.  I'm just putting the code into the repository for posterity.  

If I'm asked to complete the second portion (the insertion of users automatically into the mln database), then I'll need to worry about running the rake task.